src/vaults/SyncRequestManager.sol

```diff
@@ -275,11 +273,13 @@ contract SyncRequestManager is BaseRequestManager, ISyncRequestManager {
         balanceSheet.noteDeposit(poolId, scId, vaultDetails.asset, vaultDetails.tokenId, receiver, assets);
     }
 
-    function _maxDeposit(PoolId poolId, ShareClassId scId, address asset, uint256 tokenId)
+    function _maxDeposit(PoolId poolId, ShareClassId scId, address asset, uint256 tokenId, IBaseVault vault_)
         internal
         view
         returns (uint128 maxDeposit_)
     {
+        if (!poolManager.isLinked(poolId, scId, asset, vault_)) return 0;
```

Comment:
I think if the vault is unlinked, `maxDeposit` and `maxMint` should return 0 as you cannot deposit/mint.

Reply:
Just to be sure: For our previews, we don't factor this in right now. Is that desired or should we return 0 there as well?